### PR TITLE
Feature/195 Robustness improvements

### PR DESCRIPTION
**Pull request by lrebscher**

### CODE REVIEW

File: webapp/components/layout/LayoutManager.tsx [hunk 1]
```diff
@@ -210,21 +207,21 @@ class LayoutManager {
...
       case ADD_WIDGET: {
         this.addWidget(action.data);
         break;
       }
+
       case ADD_WIDGETS: {
         this.addWidgets(action.data);
         break;
       }
+
```
Comment (filippomc) [inline webapp/components/layout/LayoutManager.tsx:224]: Something odd with the linter

File: webapp/redux/middleware/plotMiddleware.js [hunk 2]
```diff
@@ -1,38 +1,36 @@
...
-      }
-    });
-  
+    return plotFigure(widget.id, plotMethod, plotType, store.getState().general.theme)
+      .then(result => {
+        widget.initialized = true;
+        if (result) {
+          console.debug('Plot retrieved:', widget.id);
+          widget.disabled = false;
+          widget.data = result;
```
Comment (filippomc) [inline webapp/redux/middleware/plotMiddleware.js:18]: I dont' remember what the data really is here. If it's the html, the redux store is not the right place to put it (maybe that's why we had the cache variable)

File: webapp/redux/middleware/plotMiddleware.js [hunk 3]
```diff
@@ -98,31 +101,19 @@ const plotFigure = async (plotId, plotMethod, plotType = false, theme) => {
...
-  
-
-}
-
-const setPlotToWindow = (plotId, svgResponse) => {
-  if (svgResponse === '') {
-    console.log("No plot to show")
-    return
-  }
-  window.plotSvgImages[plotId] = svgResponse
```
Comment (filippomc) [inline webapp/redux/middleware/plotMiddleware.js:127]: It's html, not svg

File: webapp/components/layout/LayoutManager.tsx (hunk 1 above)
Comment (lrebscher) [reply webapp/components/layout/LayoutManager.tsx:224]: That was me cleaning up blank lines ;)

File: webapp/redux/middleware/plotMiddleware.js (hunk 3 above)
Comment (lrebscher) [reply webapp/redux/middleware/plotMiddleware.js:127]: It was both, the name is just misleading, see how it was used in WidgetFactory.ts.

**Review by filippomc — APPROVED** [review]
Looks good!

**Pull request merged**